plugins/SlicerT/SlicerTWaveform.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * SlicerTWaveform.h - declaration of class SlicerTWaveform
+ *
+ * Copyright (c) 2023 Daniel Kauss Serna <daniel.kauss.serna@gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LMMS_SlicerT_Waveform_H
+#define LMMS_SlicerT_Waveform_H
+
+#include <QApplication>
+#include <QFontMetrics>
+#include <QInputDialog>
+#include <QMouseEvent>
+#include <QPainter>
+
+#include "Instrument.h"
+#include "SampleBuffer.h"
+
+namespace lmms {
+
+class SlicerT;
+
+namespace gui {
+
+class SlicerTWaveform : public QWidget
+{
+	Q_OBJECT
+
+public slots:
+	void updateUI();
+	void isPlaying(float current, float start, float end);
+
+public:
+	SlicerTWaveform(int w, int h, SlicerT* instrument, QWidget* parent);
+
+	// predefined sizes
+	const int m_seekerHorMargin = 5;
+	const int m_seekerHeight = 38; // used to calcualte all vertical sizes
+	const int m_middleMargin = 6;
+
+	// colors
+	const QColor s_SlicerTWaveformBgColor = QColor(255, 255, 255, 0);
+	const QColor s_SlicerTWaveformColor = QColor(123, 49, 212);
+
+	const QColor s_playColor = QColor(255, 255, 255, 200);
+	const QColor s_playHighlighColor = QColor(255, 255, 255, 70);
+
+	const QColor s_sliceColor = QColor(218, 193, 255);
+	const QColor s_selectedSliceColor = QColor(178, 153, 215);
+
+	const QColor s_seekerColor = QColor(178, 115, 255);
+	const QColor s_seekerHighlightColor = QColor(178, 115, 255, 100);
+	const QColor s_seekerShadowColor = QColor(0, 0, 0, 120);
+
+	// interaction vars
+	const float m_distanceForClick = 0.03f;
+	const float m_minSeekerDistance = 0.13f;
+	const float m_zoomSensitivity = 0.5f;
```

Comment:
```suggestion
	static constexpr float s_distanceForClick = 0.03f;
	static constexpr float s_minSeekerDistance = 0.13f;
	static constexpr float s_zoomSensitivity = 0.5f;
```
Some of these, such as `s_zoomSensitivity`, may be better off as a constexpr local variable since it's only used in one method.

Reply:
> Some of these, such as s_zoomSensitivity, may be better off as a constexpr local variable since it's only used in one method.

I think it would be preferable to have them in the header, this way you can modify most of the behavior from there.